### DISCUSSION TITLE
Prior Distributions in PHOEBE fit

### DISCUSSION THREAD

**Discussion by kittipongwangnok** [phoebe-project/phoebe2#977] [General]
I just wondering about the prior for the fitting. I found that the prior distribution is just an option, If we did not set this part then how can the emcee in PHOEBE bring prior from which part to fit the model?

**Comment by kecnry** [comment]
There are two different distribution sets defined for an MCMC sampling run:
* `init_from`: the distribution used to initialize the starting positions of the walkers
* `priors`: are folded into the log probability merit function.  See section 3 of the [inverse problem paper](https://phoebe-project.org/publications/2020Conroy+) for more details.

In addition to any user-defined priors, samples that go outside parameter bounds, fail any checks, or raise exceptions will result in a log probability of negative infinity, so you do not need to use priors to prevent these from happening.  In general, priors should only be used when they are informative and based on known/reliable information.

**Comment by kittipongwangnok** [comment]
Thank you very much for the information.
However, I got the result trapped at the same value even though we set the prior with the large range.
I've attached the result here.
How can we solve this problem? 
<img width="835" alt="Screenshot 2567-11-13 at 9 33 54 AM" src="https://github.com/user-attachments/assets/07826c68-8caf-4d35-acf6-53757eeef22a">

**Reply by kecnry** [reply]
does the resulting posterior model encompass the data appropriately?  To me it looks like the sampling is working correctly but the solution is just not constrained well - either due to the data or choice of parameterization.

**Comment by kittipongwangnok** [comment]
I have set the prior for each parameter that can draw with a large range but the result still does not convert. The prior that I set is here:
b.add_distribution({'incl@binary': phoebe.uniform(0, 90),
                    'period@binary': phoebe.uniform(100, 1000),
                    't0_supconj': phoebe.uniform(50000, 70000),
                    'teff@primary':phoebe.uniform(1000, 100000),
                   'teff@secondary':phoebe.uniform(1000, 100000),
                   'sma@binary':phoebe.uniform(1, 10000),
                    'q@binary': phoebe.uniform(0.0001, 1),
                   'requiv@primary': phoebe.uniform(1, 10000),
                   'requiv@secondary': phoebe.uniform(1, 10000)},distribution='priors_from_external_source')

and the corner plot that I got attached here as well. Could you give me some suggestions/comments? Thank you.
<img width="513" alt="Screenshot 2567-11-14 at 10 14 15 PM" src="https://github.com/user-attachments/assets/5e9bdf39-bb29-4912-b454-2cfdd66bee44">

**Reply by kecnry** [reply]
what does the logprobability vs iteration plot and model sampling the posterior look like?

**Comment by kittipongwangnok** [comment]
The plot of log probability vs iteration plot looks like this:
<img width="792" alt="Screenshot 2567-11-14 at 10 19 10 PM" src="https://github.com/user-attachments/assets/7265e11d-1691-4653-a5af-957f8f8d7e68">

**Comment by kittipongwangnok** [comment]
The model fit looks ok :)
<img width="792" alt="Screenshot 2567-11-14 at 10 22 37 PM" src="https://github.com/user-attachments/assets/a5a5744f-0bdb-4637-8dee-685e33c42ee7">

**Reply by kecnry** [reply]
can you do `sample_from` on the samples or posteriors (see [this example](https://phoebe-project.org/docs/2.3/examples/inverse_paper_examples#fig12))?  That will show how the spread in posteriors results in a spread of models relative to you data and their uncertainties.

**Comment by kittipongwangnok** [comment]
The plot from posteriors is attached here. The model fit is quite good. I was just wondering 
![Screenshot 2567-11-15 at 5 26 30 PM](https://github.com/user-attachments/assets/05a0cccf-ce0a-44b7-8f5f-82da441c2d70)
why the corner plot is not ok?

**Reply by kecnry** [reply]
What was the command you used to make that plot?  The spread looks very small given the corner plot from earlier.  Overall I think the corner plot looks reasonable, you likely just need to run more iterations.

**Comment by kittipongwangnok** [comment]
Thank you very much for replying to my message.
For the plot with the big scale, I use this command to plot: 
_ = b.plot(dataset='lc02', x='phases', model='emcee_sol_model', save = 'emcee_sol_model-lc.pdf')
_ = b.plot(dataset='lc02', x='phases', y='residuals', model='emcee_sol_model', save = 'emcee_sol_model-res.pdf')

For the small one with this command:
_ = b.plot(kind='lc', model='emcee_posts', x='phases', y='fluxes', 
           s={'dataset': 0.005},
           marker={'dataset': '.'})
_ = b.plot(kind='lc', model='emcee_posts', x='phases', y='residuals', 
           z={'dataset': 0, 'model': 1},
           save='figure_posteriors_sample_from_lc.pdf')

For more investigation, I will add more number of iterations to fit the model again.
Thank you again for your kind assistance.
